### PR TITLE
Add Atmosphere structure

### PR DESCRIPTION
**Pull request by OdNairy**
Introduce Atmosphere API

## Pull request checklist:
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add documentation comments for any added or updated public APIs.
 - [ ] Add any new public, top-level symbols to the Jazzy config's `custom_categories` (scripts/doc-generation/.jazzy.yaml)
 - [ ] Describe the changes in this PR, especially public API changes.
 - [x] Add a changelog entry to to bottom of the relevant section (typically the `## main` heading near the top).
 - [ ] Update the guides (internal access only), README.md, and DEVELOPING.md if their contents are impacted by these changes.
 - [x] Review and agree to the Contributor License Agreement ([CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement)).

Depends on:
- https://github.com/mapbox/mapbox-maps-ios/pull/1335

### CODE REVIEW

**Review by maios — APPROVED** [review]

File: Sources/MapboxMaps/Style/Style.swift [hunk 1]
```diff
@@ -930,6 +930,74 @@ public final class Style: StyleProtocol {
...
+    }
+
+    /// Gets the value of a style atmosphere property.
+    ///
+    /// - See Also: https://docs.mapbox.com/mapbox-gl-js/style-spec/fog/
+    ///
+    /// - Parameter property: Style atmosphere property name.
+    ///
+    /// - Returns: Style atmosphere property value.
+    public func atmosphereProperty(_ property: String) -> StylePropertyValue {
```
Comment (macdrevx) [inline Sources/MapboxMaps/Style/Style.swift:997]: What does it take to tell the compiler you want to use one over the other?

**Review by macdrevx — APPROVED** [review]

File: Sources/MapboxMaps/Style/Style.swift (hunk 1 above)
Comment (OdNairy) [reply Sources/MapboxMaps/Style/Style.swift:997]: I did follow Terrain API but looks like we have an issue in both Terrain and Atmosphere:

image
image

File: Sources/MapboxMaps/Style/Style.swift [hunk 2]
```diff
@@ -930,6 +930,74 @@ public final class Style: StyleProtocol {
...
+        guard let atmosphereDictionary = try atmosphere.toJSON() as? [String: Any] else {
+            throw TypeConversionError.unexpectedType
+        }
+
+        try setAtmosphere(properties: atmosphereDictionary)
+    }
+
+    /// Remove the atmosphere of the style. No fog, space or stars would be rendered.
+    public func removeAtmosphere() throws {
+        styleManager.setStyleAtmosphereForProperties(NSNull())
```
Comment (evil159) [inline Sources/MapboxMaps/Style/Style.swift:947]: Should this call be wrapped with `try handleExpected`?

Comment (OdNairy) [reply Sources/MapboxMaps/Style/Style.swift:947]: Good catch!

File: Sources/MapboxMaps/Style/Style.swift (hunk 1 above)
Comment (OdNairy) [reply Sources/MapboxMaps/Style/Style.swift]: Function that returns `Any` was removed.

**Pull request merged**